<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1180</width>
    <height>768</height>
   </rect>
  </property>
  <property name="mouseTracking">
   <bool>false</bool>
  </property>
  <property name="windowTitle">
   <string>冲压控制</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QPushButton" name="pushButtonExit">
    <property name="geometry">
     <rect>
      <x>960</x>
      <y>590</y>
      <width>121</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>退出</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButtonProc">
    <property name="geometry">
     <rect>
      <x>890</x>
      <y>30</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>冲压处理</string>
    </property>
   </widget>
   <widget class="QLabel" name="labelThresh">
    <property name="geometry">
     <rect>
      <x>900</x>
      <y>100</y>
      <width>54</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>阈值</string>
    </property>
   </widget>
   <widget class="QLabel" name="labelHoleR">
    <property name="geometry">
     <rect>
      <x>900</x>
      <y>130</y>
      <width>54</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>孔径</string>
    </property>
   </widget>
   <widget class="QSlider" name="sliderThresh">
    <property name="geometry">
     <rect>
      <x>940</x>
      <y>100</y>
      <width>160</width>
      <height>22</height>
     </rect>
    </property>
    <property name="mouseTracking">
     <bool>true</bool>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
    <property name="tickPosition">
     <enum>QSlider::NoTicks</enum>
    </property>
   </widget>
   <widget class="QSlider" name="sliderHoleR">
    <property name="geometry">
     <rect>
      <x>940</x>
      <y>130</y>
      <width>160</width>
      <height>22</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="labelThreshValue">
    <property name="geometry">
     <rect>
      <x>1110</x>
      <y>100</y>
      <width>54</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>0</string>
    </property>
   </widget>
   <widget class="QLabel" name="labelHoleRValue">
    <property name="geometry">
     <rect>
      <x>1110</x>
      <y>130</y>
      <width>54</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>0</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="checkBoxInvert">
    <property name="geometry">
     <rect>
      <x>1050</x>
      <y>20</y>
      <width>71</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>反色处理</string>
    </property>
   </widget>
   <widget class="QLabel" name="labelS2HValue">
    <property name="geometry">
     <rect>
      <x>1110</x>
      <y>160</y>
      <width>54</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>0</string>
    </property>
   </widget>
   <widget class="QSlider" name="sliderS2H">
    <property name="geometry">
     <rect>
      <x>940</x>
      <y>160</y>
      <width>160</width>
      <height>22</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="labelS2H">
    <property name="geometry">
     <rect>
      <x>890</x>
      <y>160</y>
      <width>54</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>随边距离</string>
    </property>
   </widget>
   <widget class="QLabel" name="labelH2HValue">
    <property name="geometry">
     <rect>
      <x>1110</x>
      <y>190</y>
      <width>54</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>0</string>
    </property>
   </widget>
   <widget class="QSlider" name="sliderH2H">
    <property name="geometry">
     <rect>
      <x>940</x>
      <y>190</y>
      <width>160</width>
      <height>22</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="labelH2H">
    <property name="geometry">
     <rect>
      <x>890</x>
      <y>190</y>
      <width>54</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>孔间距离</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="spinBoxX">
    <property name="geometry">
     <rect>
      <x>920</x>
      <y>260</y>
      <width>71</width>
      <height>22</height>
     </rect>
    </property>
   </widget>
   <widget class="QSpinBox" name="spinBoxY">
    <property name="geometry">
     <rect>
      <x>1050</x>
      <y>260</y>
      <width>71</width>
      <height>22</height>
     </rect>
    </property>
   </widget>
   <widget class="QSpinBox" name="spinBoxL">
    <property name="geometry">
     <rect>
      <x>920</x>
      <y>310</y>
      <width>71</width>
      <height>22</height>
     </rect>
    </property>
   </widget>
   <widget class="QSpinBox" name="spinBoxH">
    <property name="geometry">
     <rect>
      <x>1050</x>
      <y>310</y>
      <width>71</width>
      <height>22</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="labelX">
    <property name="geometry">
     <rect>
      <x>910</x>
      <y>260</y>
      <width>16</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>X</string>
    </property>
   </widget>
   <widget class="QLabel" name="labelY">
    <property name="geometry">
     <rect>
      <x>1040</x>
      <y>260</y>
      <width>16</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Y</string>
    </property>
   </widget>
   <widget class="QLabel" name="labelL">
    <property name="geometry">
     <rect>
      <x>910</x>
      <y>310</y>
      <width>16</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>L</string>
    </property>
   </widget>
   <widget class="QLabel" name="labelH">
    <property name="geometry">
     <rect>
      <x>1040</x>
      <y>310</y>
      <width>16</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>H</string>
    </property>
   </widget>
   <widget class="QLabel" name="labelUnit">
    <property name="geometry">
     <rect>
      <x>1060</x>
      <y>50</y>
      <width>71</width>
      <height>16</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Adobe Arabic</family>
      <weight>50</weight>
      <bold>false</bold>
     </font>
    </property>
    <property name="text">
     <string>单位：mm</string>
    </property>
   </widget>
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>40</y>
      <width>851</width>
      <height>661</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="tabRawVideo">
     <attribute name="title">
      <string>RawVideo</string>
     </attribute>
     <widget class="QScrollArea" name="scrollArea">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>851</width>
        <height>641</height>
       </rect>
      </property>
      <property name="widgetResizable">
       <bool>true</bool>
      </property>
      <widget class="QWidget" name="scrollAreaWidgetContents">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>849</width>
         <height>639</height>
        </rect>
       </property>
      </widget>
     </widget>
    </widget>
    <widget class="QWidget" name="tabBwVideo">
     <attribute name="title">
      <string>BwVideo</string>
     </attribute>
     <widget class="QScrollArea" name="scrollAreaBw">
      <property name="geometry">
       <rect>
        <x>-10</x>
        <y>0</y>
        <width>861</width>
        <height>641</height>
       </rect>
      </property>
      <property name="widgetResizable">
       <bool>true</bool>
      </property>
      <widget class="QWidget" name="scrollAreaWidgetContents_2">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>859</width>
         <height>639</height>
        </rect>
       </property>
       <widget class="QScrollArea" name="scrollArea_3">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>0</y>
          <width>851</width>
          <height>641</height>
         </rect>
        </property>
        <property name="widgetResizable">
         <bool>true</bool>
        </property>
        <widget class="QWidget" name="scrollAreaWidgetContents_3">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>0</y>
           <width>849</width>
           <height>639</height>
          </rect>
         </property>
        </widget>
       </widget>
      </widget>
     </widget>
    </widget>
    <widget class="QWidget" name="tabResult">
     <attribute name="title">
      <string>Result</string>
     </attribute>
     <widget class="QScrollArea" name="scrollAreaResult">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>651</width>
        <height>641</height>
       </rect>
      </property>
      <property name="widgetResizable">
       <bool>true</bool>
      </property>
      <widget class="QWidget" name="scrollAreaWidgetContents_4">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>649</width>
         <height>639</height>
        </rect>
       </property>
      </widget>
     </widget>
     <widget class="QTableWidget" name="tableResult">
      <property name="geometry">
       <rect>
        <x>650</x>
        <y>0</y>
        <width>201</width>
        <height>641</height>
       </rect>
      </property>
      <column>
       <property name="text">
        <string>X</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Y</string>
       </property>
      </column>
     </widget>
    </widget>
   </widget>
   <widget class="QLabel" name="labelROIReal">
    <property name="geometry">
     <rect>
      <x>890</x>
      <y>230</y>
      <width>81</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>ROI实际尺寸</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="spinBoxTop">
    <property name="geometry">
     <rect>
      <x>990</x>
      <y>360</y>
      <width>51</width>
      <height>31</height>
     </rect>
    </property>
    <property name="maximum">
     <number>3000</number>
    </property>
   </widget>
   <widget class="QSpinBox" name="spinBoxButton">
    <property name="geometry">
     <rect>
      <x>990</x>
      <y>480</y>
      <width>51</width>
      <height>31</height>
     </rect>
    </property>
    <property name="maximum">
     <number>3000</number>
    </property>
   </widget>
   <widget class="QSpinBox" name="spinBoxLeft">
    <property name="geometry">
     <rect>
      <x>910</x>
      <y>420</y>
      <width>51</width>
      <height>31</height>
     </rect>
    </property>
    <property name="maximum">
     <number>3000</number>
    </property>
   </widget>
   <widget class="QSpinBox" name="spinBoxRight">
    <property name="geometry">
     <rect>
      <x>1070</x>
      <y>420</y>
      <width>51</width>
      <height>31</height>
     </rect>
    </property>
    <property name="maximum">
     <number>3000</number>
    </property>
   </widget>
   <widget class="QLabel" name="labelROIEdge">
    <property name="geometry">
     <rect>
      <x>890</x>
      <y>360</y>
      <width>81</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>ROI边界</string>
    </property>
   </widget>
   <widget class="Line" name="line">
    <property name="geometry">
     <rect>
      <x>890</x>
      <y>340</y>
      <width>241</width>
      <height>20</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="Line" name="line_2">
    <property name="geometry">
     <rect>
      <x>890</x>
      <y>210</y>
      <width>241</width>
      <height>20</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>1000</x>
      <y>390</y>
      <width>41</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>上边界</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>1000</x>
      <y>450</y>
      <width>51</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>下高度</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>960</x>
      <y>420</y>
      <width>41</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>左边界</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>1030</x>
      <y>420</y>
      <width>41</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>右宽度</string>
    </property>
   </widget>
   <widget class="QSlider" name="sliderFrameRate">
    <property name="geometry">
     <rect>
      <x>580</x>
      <y>0</y>
      <width>160</width>
      <height>22</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="labelFrameRate">
    <property name="geometry">
     <rect>
      <x>513</x>
      <y>0</y>
      <width>81</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>帧间隔(ms)</string>
    </property>
   </widget>
   <widget class="QLabel" name="labelFrameRateValue">
    <property name="geometry">
     <rect>
      <x>750</x>
      <y>0</y>
      <width>21</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>30</string>
    </property>
   </widget>
   <widget class="QLabel" name="labelHough">
    <property name="geometry">
     <rect>
      <x>520</x>
      <y>30</y>
      <width>81</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>HoughValue</string>
    </property>
   </widget>
   <widget class="QSlider" name="sliderHoughValue">
    <property name="geometry">
     <rect>
      <x>580</x>
      <y>30</y>
      <width>160</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <number>100</number>
    </property>
    <property name="maximum">
     <number>180</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="labelHoughValue">
    <property name="geometry">
     <rect>
      <x>750</x>
      <y>30</y>
      <width>21</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>150</string>
    </property>
   </widget>
   <widget class="QComboBox" name="comboBoxAlgoChoose">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>10</y>
      <width>141</width>
      <height>22</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="labelAlgoTime">
    <property name="geometry">
     <rect>
      <x>230</x>
      <y>10</y>
      <width>91</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Algorithm Time: </string>
    </property>
   </widget>
   <widget class="QLabel" name="labelAlgoTimeValue">
    <property name="geometry">
     <rect>
      <x>330</x>
      <y>10</y>
      <width>81</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>0 ms</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_Algotype">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>54</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>排孔方式</string>
    </property>
   </widget>
   <zorder>tabWidget</zorder>
   <zorder>pushButtonExit</zorder>
   <zorder>pushButtonProc</zorder>
   <zorder>labelThresh</zorder>
   <zorder>labelHoleR</zorder>
   <zorder>sliderThresh</zorder>
   <zorder>sliderHoleR</zorder>
   <zorder>labelThreshValue</zorder>
   <zorder>labelHoleRValue</zorder>
   <zorder>checkBoxInvert</zorder>
   <zorder>labelS2HValue</zorder>
   <zorder>sliderS2H</zorder>
   <zorder>labelS2H</zorder>
   <zorder>labelH2HValue</zorder>
   <zorder>sliderH2H</zorder>
   <zorder>labelH2H</zorder>
   <zorder>spinBoxX</zorder>
   <zorder>spinBoxY</zorder>
   <zorder>spinBoxL</zorder>
   <zorder>spinBoxH</zorder>
   <zorder>labelX</zorder>
   <zorder>labelY</zorder>
   <zorder>labelL</zorder>
   <zorder>labelH</zorder>
   <zorder>labelUnit</zorder>
   <zorder>labelROIReal</zorder>
   <zorder>spinBoxTop</zorder>
   <zorder>spinBoxButton</zorder>
   <zorder>spinBoxLeft</zorder>
   <zorder>spinBoxRight</zorder>
   <zorder>labelROIEdge</zorder>
   <zorder>line</zorder>
   <zorder>line_2</zorder>
   <zorder>label</zorder>
   <zorder>label_2</zorder>
   <zorder>label_3</zorder>
   <zorder>label_4</zorder>
   <zorder>sliderFrameRate</zorder>
   <zorder>labelFrameRate</zorder>
   <zorder>labelFrameRateValue</zorder>
   <zorder>labelHough</zorder>
   <zorder>sliderHoughValue</zorder>
   <zorder>labelHoughValue</zorder>
   <zorder>comboBoxAlgoChoose</zorder>
   <zorder>labelAlgoTime</zorder>
   <zorder>labelAlgoTimeValue</zorder>
   <zorder>label_Algotype</zorder>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1180</width>
     <height>23</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
